tests/Cache/FileCacheTest.php

```diff
@@ -87,21 +89,50 @@ public function testMissingCacheFileRead(): void
         $cache = $this->createCache();
 
         $this->expectException(CacheException::class);
-        $this->expectExceptionMessageMatches('/^Can not load cache: File "[^"]+" cannot be read: .+$/');
+        $this->expectExceptionMessageMatches('/^Can not load cache: File ".+" cannot be read: .+$/');
 
         $cache->read('missing_file');
     }
 
-    public function testUnwriteableCacheDir(): void
+    public function testUnwritableCachePath(): void
+    {
+        $root = vfsStream::setup();
+        $root->addChild(new vfsStreamDirectory($this->getCacheDirName(), 0));
+
+        $this->expectExceptionMessageMatches('/^Cache path ".+" is not writeable\. Check your filesystem permissions or disable Gherkin file cache\.$/');
+        $this->expectException(CacheException::class);
+
+        new FileCache($root->url());
+    }
+
+    public function testUncreatableCachePath(): void
     {
         $root = vfsStream::setup();
         $root->chmod(0);
 
+        $this->expectExceptionMessageMatches('/^Cache path ".+" cannot be created or is not a directory: .+ cannot be created\.$/');
         $this->expectException(CacheException::class);
 
         new FileCache($root->url());
     }
 
+    public function testNonDirectoryCachePath(): void
+    {
+        $root = vfsStream::setup();
+        $directory = new class('some-dir', 0777) extends vfsStreamDirectory {
+            public function addChild(vfsStreamContent $child): void
+            {
+                trigger_error('Test warning', E_USER_WARNING);
```

Comment:
I'm not a fan of this, but I couldn't find any other way to make `mkdir()` trigger an error "natively".